include/zenoh-pico/collections/pointer.h

```diff
@@ -73,9 +86,15 @@
     static inline name##_sptr_t name##_sptr_new(type##_t val) {                                 \
         name##_sptr_t p;                                                                        \
         p.ptr = (type##_t *)z_malloc(sizeof(type##_t));                                         \
-        *p.ptr = val;                                                                           \
         p._cnt = (uint8_t *)z_malloc(sizeof(uint8_t));                                          \
```

Comment:
This malloc should be inside the if (p.ptr != NULL). If for any obscure reason the first malloc fails and the second one succeeds, we have a memory leak because the corresponding cleanup is not done.

Reply:
Done in the upcoming commits.